src/utils.ts

```diff
@@ -135,6 +137,43 @@ function words<T extends string>(sentence: T): Words<T> {
     .split(/\s+/g) as Words<T>
 }
 
+/**
+ * Truncate a string if it's longer than the given maximum length.
+ * The last characters of the truncated string are replaced with the omission string which defaults to "...".
+ */
+type Truncate<
+  T extends string,
+  Size extends number,
+  Omission extends string = '...',
+> = Math.IsNegative<Size> extends true
+  ? Omission
+  : Math.Subtract<Split<T>['length'], Size> extends 0
+  ? T
+  : `${Slice<T, 0, Math.Subtract<Size, Split<Omission>['length']>>}${Omission}`
```

Comment:
```suggestion
  : Join<[Slice<T, 0, Math.Subtract<Size, Length<Omission>>>, Omission]>
```

Reply:
fixed in [f155f9b](https://github.com/gustavoguichard/string-ts/pull/41/commits/f155f9b3c532da62dd2ca2ec26686467f3e267f3)